src/main/mw.c

```diff
@@ -291,10 +291,11 @@ void mwDisarm(void)
 #define TELEMETRY_FUNCTION_MASK (FUNCTION_TELEMETRY_FRSKY | FUNCTION_TELEMETRY_HOTT | FUNCTION_TELEMETRY_SMARTPORT | FUNCTION_TELEMETRY_LTM)
 
 void releaseSharedTelemetryPorts(void) {
-    serialPort_t *sharedPort = findSharedSerialPort(TELEMETRY_FUNCTION_MASK, FUNCTION_MSP);
-    while (sharedPort) {
-        mspReleasePortIfAllocated(sharedPort);
-        sharedPort = findNextSharedSerialPort(TELEMETRY_FUNCTION_MASK, FUNCTION_MSP);
+    for(serialPort_t *sharedPort = findSharedSerialPort(TELEMETRY_FUNCTION_MASK, FUNCTION_MSP);
+        sharedPort;
+        sharedPort = findNextSharedSerialPort(TELEMETRY_FUNCTION_MASK, FUNCTION_MSP)
+        ) {
+        mspSerialReleasePortIfAllocated(sharedPort);
```

Comment:
I'll revert it ..
But the loop is 'for' loop - initializatiopn part / test / increment

The whole concept of non-reentrant find/findNext function is terrible. 
What about modelling it like 'PG_FOREACH' and 'PG_FOREACH_PROFILE' (iterate all serial ports and continue on ports not matching user criteria)?
Or use `serialPort_t*` as state for iteration - but it will be necessary to pass the condition twice.
Or implement full iterator with condition stored in (explicit) state.